typeinfo-less.md

```diff
@@ -0,0 +1,305 @@
+```
+
+It is worth noting that both signedness and size modifiers only exist on native
+types, and as such will only be taken into account when sorting user defined
+types. 
+
+### Ordering References and Pointers
+
+Pointers, lvalue references, and rvalue references should be ordered as 
+follows:
+
+```cpp
+// ------ Lowest ------
+using T = ...;
+
+T // unspecified
+T* // Pointer
+T& // lvalue Reference
+T&& // rvalue Reference
+
+// ------ Highest ------
+```
+
+### Type Qualifiers
+
+`const`, `volatile` and `mutable` qualifiers should be ordered as follows:
+
+Volatile modifiers would be ordered with `unspecified < volatile`.
+
+Constantness modifiers would be ordered: `unspecified < const < mutable`
+
+Order first based on volatileness, then on constantness.
+
+For example:
+
+```cpp
+// ------ Lowest ------
+
+using T = ...;
+
+T
+const T
+mutable T
+const volatile T
+mutable volatile T
+
+// ------ Highest ------
+```
+
+### Non Templated User Defined Types
+
+Simple User Defined Types can be defines as either a `struct` or a `class` that
+has been declared without a template parameter. As far as ordering is concerned,
+structs and classes should be treated the same.
+
+User defined types will be greater than native types.
+
+Ordering these is simple, since we cannot redeclare two types with the same
+name, we'll just order simple user defined types alphabetically. 
+
+```cpp
+struct Apple {};
+class Banana {};
+struct Carrot {};
+
+```
+
+Would be ordered as `Apple < Banana < Carrot`.
+
+References and pointers to, as well as type qualifiers on, will behave the same
+way on user defined types as they do on native types. For example, given:
+
+```cpp
+Apple
+Carrot
+Apple&
+Banana const&
+Carrot&&
+```
+
+Would be ordered `Apple < Apple& < Banana const&  < Carrot < Carrot&&`
+
+### namespaces
+
+Namespaces will also be ordered alphabetically.
+
+Given:
+```cpp
+namespace foo {
+  struct i;
+}
+
+namespace bar {
+  struct i;
+}
+```
+
+The order of the two structs w/ type `i` types would be `bar::i < foo::i`
+
+Combining the above, we can define an order for every possible combination of
+type, qualifier, and pointer/reference. 
+
+1. Order alphabetically by namespace and type. Native types come before user defined types.
+2. Order by pointer or reference. 
+3. Order by type qualifier.
+
+### Type Aliases 
+
+The ordering of type aliases is to be the same as the type they are aliased to.
+
+For example, the order of the following `<cstdint>` type aliases would be:
+
+```cpp
+// ------ Lowest ------
+
+int8_t // signed char
+uint8_t // unsigned char
+int16_t // signed short int
+uint16_t // unsigned short int
+int32_t // long int
+uint32_t // unsigned long int
+int64_t // long long int
+uint64_t // unsigned long long int
+
+// ------ Highest ------
+```
+
+## Non-Templated User Defined Functions
+
+User Defined function types will be ordered by 
+
+1. Name (alphabetically)
+2. Return type, ordered as described above
```

Comment:
I think return-type is redundant; once you order by parameter count and then by parameter type, return types are unique.

Reply:
Actually i think i might be an idiot.

Type aliases can still be template arguments. Jesus. We need to order them.